lesson_09/oop/oop_app/src/test/java/com/codedifferently/lesson9/JadasLibraryTest/ConcretePatronTest.java

```diff
@@ -0,0 +1,36 @@
+package com.codedifferently.lesson9.JadasLibraryTest;
+
+import com.codedifferently.lesson9.JadasLibrary.Book;
+import com.codedifferently.lesson9.JadasLibrary.Patron;
+import java.util.ArrayList;
+import java.util.List;
+
+public class ConcretePatronTest implements Patron {
```

Comment:
This is what we call a stub in testing. It would have been better for you to use a mock in this case as it would require less code and, more importantly, less implementation logic.

This really should have been your Patron implementation. I would've suggested renaming this to `PatronImpl` and moving into the main source code. I'll pretend you did that this time :D